### PR TITLE
added an example of copying artifact manifest

### PR DESCRIPTION
**Pull request by wangxiaoxuan273**
Signed-off-by: wangxiaoxuan273 <wangxiaoxuan119@gmail.com>

Part of issue #178

### CODE REVIEW

File: example_test.go [hunk 1]
```diff
@@ -252,3 +280,24 @@ func ExampleCopy_localToRemote() {
...
+	ctx := context.Background()
+	err = oras.CopyGraph(ctx, src, dst, exampleSignatureManifestDescriptor, oras.DefaultCopyGraphOptions)
+	if err != nil {
+		panic(err)
+	}
+
+	// verify that the content of dst is described by exampleSignatureManifestDescriptor
+	contentExists, err := dst.Exists(ctx, exampleSignatureManifestDescriptor)
+	fmt.Println(contentExists)
+
```
Comment (wangxiaoxuan273) [inline example_test.go:300]: I think we need a better way to verify the copied content and its subject, but I don't know how.

File: example_test.go [hunk 2]
```diff
@@ -252,3 +280,24 @@ func ExampleCopy_localToRemote() {
...
+func ExampleCopy_artifactManifestRemoteToLocal() {
+	src, err := remote.NewRepository(fmt.Sprintf("%s/%s", remoteHost, "source"))
+	if err != nil {
+		panic(err)
+	}
+	dst := memory.New()
+
+	ctx := context.Background()
+	err = oras.CopyGraph(ctx, src, dst, exampleSignatureManifestDescriptor, oras.DefaultCopyGraphOptions)
```
Comment (wangxiaoxuan273) [inline example_test.go:292]: Should we use `Copy`, `ExtendedCopy`, `CopyGraph` or `ExtendedCopyGraph` here?

Comment (Wwwsylvia) [reply example_test.go:292]: I think we can resolve by a given digest, and then call `CopyGraph()`.

File: example_test.go (hunk 1 above)
Comment (Wwwsylvia) [reply example_test.go:300]: We can follow the same pattern as other examples.

File: example_test.go (hunk 2 above)
Comment (wangxiaoxuan273) [reply example_test.go:292]: Changed accordingly.

File: example_test.go [hunk 3]
```diff
@@ -252,3 +280,30 @@ func ExampleCopy_localToRemote() {
 	// Output:
 	// sha256:6f2590d54af17afaca41a6243e3c01b368117d24b32e7581a6dee1d856dd3c4b
 }
+
+func ExampleCopy_artifactManifestRemoteToLocal() {
```
[the remark this answers is not in the document]
Comment (wangxiaoxuan273) [reply example_test.go]: Added doc comment.

File: example_test.go [hunk 4]
```diff
@@ -252,3 +280,35 @@ func ExampleCopy_localToRemote() {
...
+// an artifact manifest from remote location to local.
+func Example_copyArtifactManifestRemoteToLocal() {
+	src, err := remote.NewRepository(fmt.Sprintf("%s/%s", remoteHost, "source"))
+	if err != nil {
+		panic(err)
+	}
+	dst := memory.New()
+	ctx := context.Background()
+
+	// The digest can be obtained from the Referrers API.
```
[the remark this answers is not in the document]
Comment (wangxiaoxuan273) [reply example_test.go:294]: deleted.

File: example_test.go [hunk 5]
```diff
@@ -252,3 +280,35 @@ func ExampleCopy_localToRemote() {
...
+	if err != nil {
+		panic(err)
+	}
+	err = oras.CopyGraph(ctx, src, dst, exampleDescriptor, oras.DefaultCopyGraphOptions)
+	if err != nil {
+		panic(err)
+	}
+
+	// verify that the content of dst is described by exampleSignatureManifestDescriptor
+	contentExists, err := dst.Exists(ctx, exampleSignatureManifestDescriptor)
```
[the remark this answers is not in the document]
Comment (wangxiaoxuan273) [reply example_test.go:306]: changed.

File: example_test.go [hunk 6]
```diff
@@ -252,3 +280,34 @@ func ExampleCopy_localToRemote() {
 	// Output:
 	// sha256:6f2590d54af17afaca41a6243e3c01b368117d24b32e7581a6dee1d856dd3c4b
 }
+
+// Example_copyArtifactManifestRemoteToLocal gives an example of copying
+// an artifact manifest from remote location to local.
```
Comment (Wwwsylvia) [inline example_test.go:285]: nit: How about
```suggestion
// an artifact manifest from a remote repository to local.
```

File: example_test.go [hunk 7]
```diff
@@ -252,3 +280,34 @@ func ExampleCopy_localToRemote() {
...
+	exampleDescriptor, err := src.Resolve(ctx, exampleDigest)
+	if err != nil {
+		panic(err)
+	}
+	err = oras.CopyGraph(ctx, src, dst, exampleDescriptor, oras.DefaultCopyGraphOptions)
+	if err != nil {
+		panic(err)
+	}
+
+	// verify that the content of dst is described by exampleSignatureManifestDescriptor
```
Comment (Wwwsylvia) [inline example_test.go:304]: nit: verify that the artifact manifest described by the descriptor exists in dst

File: example_test.go [hunk 8]
```diff
@@ -99,6 +117,16 @@ func TestMain(m *testing.M) {
 			w.WriteHeader(http.StatusAccepted)
 		case strings.Contains(p, "/blobs/uploads/"+exampleUploadUUid) && m == "GET":
 			w.WriteHeader(http.StatusCreated)
+		case strings.Contains(p, string(exampleSignatureManifestDescriptor.Digest)):
```
[the remark this answers is not in the document]
Comment (wangxiaoxuan273) [reply example_test.go:120]: Added the check.

**Review by Wwwsylvia — APPROVED** [review]
LGTM

File: example_test.go [hunk 9]
```diff
@@ -252,3 +280,34 @@ func ExampleCopy_localToRemote() {
 	// Output:
 	// sha256:6f2590d54af17afaca41a6243e3c01b368117d24b32e7581a6dee1d856dd3c4b
 }
+
+// Example_copyArtifactManifestRemoteToLocal gives an example of copying
+// an artifact manifest from a remote repository to local.
+func Example_copyArtifactManifestRemoteToLocal() {
+	src, err := remote.NewRepository(fmt.Sprintf("%s/%s", remoteHost, "source"))
```
Comment (qweeah) [inline example_test.go:287]:
```suggestion
	src, err := remote.NewRepository(fmt.Sprintf("%s/source", remoteHost))
```

**Review by qweeah — APPROVED** [review]
LGTM

**Pull request merged**